sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -73,33 +77,25 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] {
               postScanFilters)
             Aggregate(groupingExpressions, resultExpressions, plan)
           } else {
-            val resultAttributes = resultExpressions.map(_.toAttribute)
-              .map ( e => e match { case a: AttributeReference => a })
-            var index = 0
             val aggOutputBuilder = ArrayBuilder.make[AttributeReference]
-            for (a <- resultAttributes) {
-              val newName = if (a.name.contains("FILTER")) {
-                a.name.substring(0, a.name.indexOf("FILTER") - 1)
-              } else if (a.name.contains("DISTINCT")) {
-                a.name.replace("DISTINCT ", "")
-              } else {
-                a.name
-              }
-
-              aggOutputBuilder +=
-                a.copy(name = newName,
-                  dataType = aggregates(index).dataType)(exprId = NamedExpression.newExprId,
-                  qualifier = a.qualifier)
-              index += 1
+            for (a <- aggregates) {
+                aggOutputBuilder += AttributeReference(toPrettySQL(a), a.dataType)()
             }
             val aggOutput = aggOutputBuilder.result
 
-            var newOutput = aggOutput
-            for (col <- output) {
-              if (!aggOutput.exists(_.name.contains(col.name))) {
-                newOutput = col +: newOutput
+            val newOutputBuilder = ArrayBuilder.make[AttributeReference]
+            for (col1 <- output) {
+              var found = false
+              for (col2 <- aggOutput) {
+                  if (contains(col2.name, col1.name)) {
```

Comment:
Thanks very much for the fix !  We have evaluated it and it is working great for us !  
We do have one minor nit. 
We noticed that very short column names like col = 'i' can end up with an exception, because the compare can mistakenly match portions of the aggregate expression for short column names.
For example, the column name col= 'i' can match with col2 = "min(k)".
The following fix seems to solve the issue for us:
if (col2.name.toLowerCase(Locale.ROOT).contains("(" + col1.name.toLowerCase(Locale.ROOT))) {
The single "(" came about since we saw cases where col2 looked like: "sum(CAST(I AS BIGINT))#15" or in other cases col2 was something like this: "min(i)"

Reply:
Thanks @rf972. I committed the fix. 
Are there any performance improvement with aggregate push down in your TPCH tests?

Reply:
Thanks @huaxingao for incorporating our changes in the patch !   
Here is a brief overview of our TPCH testing with this patch.   We initially picked two queries from TPCH which should show immediate benefits from aggregate pushdown.  The Q01 test was our first choice, but we put that aside for now since it has UDFs and the count operator.  So now we are focusing on the Q06 test.  Our simulations show that the filter pushdown in Q06 reduces the data transfer (from storage/database) significantly (in our simulated case from 700 MB down to about 1.5 MB).  Our simulations also show that aggregate pushdown further reduces the data transfer from that 1.5 MB down to just 34 bytes.  
The Q06 TPCH test requires support for an expression under an aggregate.  So something like sum(x * y).  We are actively working on this now, and once we finish adding this support (within a week or so, we hope), we will share the code and our actual TPCH results.  Also if you have any thoughts on the direction for adding this support that would be appreciated too.  Thanks!

Reply:
Thanks @huaxingao  for incorporating our suggestions into the patch regarding aggregates containing expressions!  
With your changes, TPCH Q6 test shows a substantial reduction in data transfer.  
With just filter and projection pushdown, Q6 transfer size is about 1.5 MB.  But with aggregate pushdown this now gets reduced to 17 bytes.

In our testing we found a case that throws an error.  Grouping by more than one column seems to cause an error.  Here is an example that fails for us:

```
val df = sparkSession.sql("select BONUS, SUM(SALARY+BONUS), SALARY FROM h2.test.employee" + 
       " GROUP BY SALARY, BONUS")
df.show()
```

If you have any thoughts or suggestions on a solution for this, we would appreciate it.  Thanks !

Reply:
@rf972 Thanks for sharing the TCPH tests results! I will fix the bug. I actually haven't done extensive tests yet.  I will do more tests after the Thanksgiving weekend.